framework/src/base/Assembly.C

```diff
@@ -1522,6 +1524,19 @@ Assembly::addCachedJacobianContributions(SparseMatrix<Number> & jacobian)
 }
 
 void
+Assembly::registerUserObjectShapeVariable(unsigned int var)
+{
+  std::vector<unsigned int> & reg = _user_object_shape_variables;
+
+  // add variable number to the list of registered variables
+  reg.push_back(var);
+
+  // sort and keep elements unique
+  std::sort(reg.begin(), reg.end());
+  reg.erase(std::unique(reg.begin(), reg.end()), reg.end());
```

Comment:
Iterating over a `set` with an iterator is going to be WAY in the noise.  When you compare an iterator movement and dereference to incrementing a local counter and then indexing into the vector (like you are doing now) there's no way there's any difference that would be measureable in MOOSE.

A `set` shows everyone exactly what this is: a unique list of IDs.  No one can ever screw it up by "side loading" values into it... and it's self documenting.